.github/workflows/check-main-branch-ci.yaml

```diff
@@ -0,0 +1,58 @@
+name: Check main branch status
+
+on:
+  pull_request:
+
+jobs:
+  block-release-if-main-broken:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          # Fetch all history so we can access the base branch commit
+          fetch-depth: 0
+
+      - name: Get base version
+        # Pull the Chart.yaml from the base commit, then extract `.version` using yq
+        run: |
+          base_version=$(git show origin/${{ github.event.pull_request.base.ref }}:helm-charts/splunk-otel-collector/Chart.yaml \
+            | yq e '.version' -)
+          echo "base_version=$base_version" >> $GITHUB_ENV
+
+      - name: Get PR version
+        # Extract `.version` from the PR's current Chart.yaml
+        run: |
+          pr_version=$(yq e '.version' helm-charts/splunk-otel-collector/Chart.yaml)
+          echo "pr_version=$pr_version" >> $GITHUB_ENV
+
+      - name: Compare versions
+        id: compare
+        run: |
+          echo "Base version: ${{ env.base_version }}"
+          echo "PR version: ${{ env.pr_version }}"
+
+          if [ "${{ env.base_version }}" = "${{ env.pr_version }}" ]; then
+            echo "version_changed=false" >> $GITHUB_OUTPUT
+            echo "No version change detected — skipping main-branch check."
+          else
+            echo "version_changed=true" >> $GITHUB_OUTPUT
+            echo "Version changed!"
+          fi
+
+      - name: Check main branch build status
+        # Only run this step if the version actually changed
+        if: steps.compare.outputs.version_changed == 'true'
+        run: |
+          response=$(curl -s \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            "https://api.github.com/repos/${{ github.repository }}/commits/main/status")
```

Comment:
Do you know if this gets the status of the last completed run? or of runs in progress too? The workflow run endpoint gives a status and conclusion to differentiate this https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks#check-statuses-and-conclusions. Not sure if this is a concern here.

Reply:
Good question. I don't know. Looks like it should be _any_ of the last commit including in_progress. But let me print the fetched status so we can easy debug if needed